src/plugins/orangehrmCorporateDirectoryPlugin/Dao/EmployeeDirectoryDao.php

```diff
@@ -0,0 +1,121 @@
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\ORM\QueryBuilderWrapper;
+
+class EmployeeDirectoryDao extends BaseDao
+{
+    use TextHelperTrait;
+
+    /**
+     * @param EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder
+     * @return Employee[]
+     */
+    public function getEmployeeList(EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder): array
+    {
+        $qb = $this->getEmployeeListQueryBuilderWrapper($employeeDirectorySearchParamHolder)->getQueryBuilder();
+        return $qb->getQuery()->execute();
+    }
+
+    /**
+     * @param EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder
+     * @return QueryBuilderWrapper
+     */
+    protected function getEmployeeListQueryBuilderWrapper(
+        EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder
+    ): QueryBuilderWrapper {
+        $q = $this->createQueryBuilder(Employee::class, 'employee');
+        $q->leftJoin('employee.jobTitle', 'jobTitle');
+        $q->leftJoin('employee.subDivision', 'subunit');
+        $q->leftJoin('employee.locations', 'location');
+
+        $this->setSortingAndPaginationParams($q, $employeeDirectorySearchParamHolder);
+
+        if (is_null($employeeDirectorySearchParamHolder->getIncludeEmployees()) ||
+            $employeeDirectorySearchParamHolder->getIncludeEmployees() ===
+            EmployeeDirectorySearchFilterParams::INCLUDE_EMPLOYEES_ONLY_CURRENT
+        ) {
+            $q->andWhere($q->expr()->isNull('employee.employeeTerminationRecord'));
+        } elseif (
+            $employeeDirectorySearchParamHolder->getIncludeEmployees() ===
+            EmployeeDirectorySearchFilterParams::INCLUDE_EMPLOYEES_ONLY_PAST
+        ) {
+            $q->andWhere($q->expr()->isNotNull('employee.employeeTerminationRecord'));
+        }
+
+        if (!is_null($employeeDirectorySearchParamHolder->getEmpNumber())) {
+            $q->andWhere('employee.empNumber = :empNumber')
+                ->setParameter('empNumber', $employeeDirectorySearchParamHolder->getEmpNumber());
+        }
+
+        if (!is_null($employeeDirectorySearchParamHolder->getLocationId())) {
+            $q->andWhere('location.id = :locationId')
+                ->setParameter('locationId', $employeeDirectorySearchParamHolder->getLocationId());
+        }
+
+        if (!is_null($employeeDirectorySearchParamHolder->getJobTitleId())) {
+            $q->andWhere('jobTitle.id = :jobTitleId')
+                ->setParameter('jobTitleId', $employeeDirectorySearchParamHolder->getJobTitleId());
+        }
+
+        $q->andWhere($q->expr()->isNull('employee.purgedAt'));
+
+        return $this->getQueryBuilderWrapper($q);
+    }
+
+    /**
+     * @param EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder
+     * @return int[]
+     */
+    public function getEmpNumbersByFilterParams(EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder
```

Comment:
any use of this method? if not using remove this

Reply:
Done